lib/all_changed_attributes.rb

```diff
@@ -0,0 +1,47 @@
+require 'all_changed_attributes/version'
+
+module AllChangedAttributes
+  def write_attribute(attr_name, value)
+    store_all_changed_attributes(attr_name, value) if attributes[attr_name] != value
+    super
+  end
+
+  def all_changed_attributes(new = {})
```

Comment:
I think the `new` argument would be confusing for a caller.  I think you should just change `all_changed_attributes(...)` calls into `all_changed_attributes.merge(...)` if they want to merge in attributes